[FEATURE] - Have cached copies of tools included

### Which component(s) does this affect?

- [ ] Full Dashboard
- [ ] Lite
- [ ] SQL collection scripts
- [x] Installer
- [ ] Documentation

### Problem Statement

Just installed the full version of the Performance Monitor tool and loving every bit of it!

But the tool was installed on a mgmt server, that then connects to and SQL server - but neither the mgmt server and the SQL server have internet access, so it fails to download some of the tools (DarlingData, First responder kit, etc).

```
OK] 51_collect_running_jobs.sql - Success
Executing 52_collect_database_size_stats.sql...
[OK] 52_collect_database_size_stats.sql - Success
Executing 53_collect_server_properties.sql...
[OK] 53_collect_server_properties.sql - Success
Executing 54_create_finops_views.sql...
[OK] 54_create_finops_views.sql - Success
Installing community dependencies...
Installing community dependencies...
Installing sp_WhoIsActive...
[ERROR] sp_WhoIsActive - Failed: The request was canceled due to the configured HttpClient.Timeout of 30 seconds elapsing.
Installing DarlingData...
[ERROR] DarlingData - Failed: The request was canceled due to the configured HttpClient.Timeout of 30 seconds elapsing.
Installing First Responder Kit...
[ERROR] First Responder Kit - Failed: The request was canceled due to the configured HttpClient.Timeout of 30 seconds elapsing.
[WARN] Dependencies installed: 0/3
Executing 98_validate_installation.sql...
[OK] 98_validate_installation.sql - Success
Recording installation history...
[OK] Installation history recorded
Running post-install validation...
Running initial collection to validate installation...
Executing master collector...
[OK] Master collector completed
[WARN] Validation complete: 42 collectors succeeded, 6 failed
[ERROR]   process_blocked_process_xml: Error processing blocked process XML: sp_HumanEventsBlockViewer is not installed in PerformanceMonitor or master. Please install it from https://github.com/erikdarlingdata/DarlingData
[ERROR]   process_blocked_process_xml: sp_HumanEventsBlockViewer is not installed in PerformanceMonitor or master. Please install it from https://github.com/erikdarlingdata/DarlingData
[ERROR]   system_health_collector: Error in system health collector wrapper: sp_HealthParser not found in PerformanceMonitor or master - cannot collect system health data
[ERROR]   system_health_collector: sp_HealthParser not found in PerformanceMonitor or master - cannot collect system health data
[ERROR]   query_snapshots_collector: Error in query snapshots collector: sp_WhoIsActive is not installed in PerformanceMonitor or master. Please install it from https://github.com/amachanic/sp_whoisactive
[ERROR]   query_snapshots_collector: sp_WhoIsActive is not installed in PerformanceMonitor or master. Please install it from https://github.com/amachanic/sp_whoisactive
[WARN] Validation: 42 collectors succeeded, 6 failed
```

### Proposed Solution

In each of the versions of the Performance Monitor tool, could we have a cached version of the newest version of DarlingData and FirstResponder kit included, so that it can be installed without any internet access?

### Use Case

Allows us to install / configure a SQL server to be monitored, even if neither the mgmt server and the SQL server, dont have any internet access.

### Alternatives Considered

_No response_

### Additional Context

Just wanted to say thanks! The tool seems awesome, and its just the right tool for daily use cases - no bloat!

--- a/.gitignore
+++ b/.gitignore
@@ -59,3 +59,6 @@ Lite/collection_schedule.json
 
 # Plans directory
 plans/
+
+# Community scripts (user-provided, not bundled)
+community/*.sql

--- a/Dashboard/AddServerDialog.xaml.cs
+++ b/Dashboard/AddServerDialog.xaml.cs
@@ -618,7 +618,8 @@ namespace PerformanceMonitorDashboard
                     preValidationAction = async () =>
                     {
                         AppendInstallLog("Installing community dependencies...", "Info");
-                        using var depInstaller = new DependencyInstaller();
+                        string communityDir = System.IO.Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "community");
+                        using var depInstaller = new DependencyInstaller(communityDir);
                         await depInstaller.InstallDependenciesAsync(installerConnStr, progress, cancellationToken);
                     };
                 }

--- a/Installer.Core/DependencyInstaller.cs
+++ b/Installer.Core/DependencyInstaller.cs
@@ -14,23 +14,31 @@ namespace Installer.Core;
 
 /// <summary>
 /// Installs community dependencies (sp_WhoIsActive, DarlingData, First Responder Kit)
-/// from GitHub. Requires an HttpClient — create one instance and dispose when done.
+/// from a local community/ directory or GitHub. Local files are checked first — if
+/// present, the network is not used. This supports air-gapped installations.
 /// </summary>
 public sealed class DependencyInstaller : IDisposable
 {
     private readonly HttpClient _httpClient;
+    private readonly string? _communityDirectory;
     private bool _disposed;
 
-    public DependencyInstaller()
+    /// <param name="communityDirectory">
+    /// Optional path to a community/ directory containing pre-downloaded SQL files.
+    /// When provided and files exist, they are used instead of downloading from GitHub.
+    /// </param>
+    public DependencyInstaller(string? communityDirectory = null)
     {
         _httpClient = new HttpClient
         {
             Timeout = TimeSpan.FromSeconds(30)
         };
+        _communityDirectory = communityDirectory;
     }
 
     /// <summary>
-    /// Install community dependencies from GitHub into the PerformanceMonitor database.
+    /// Install community dependencies into the PerformanceMonitor database.
+    /// Checks the community/ directory first, falls back to GitHub download.
     /// Returns the number of successfully installed dependencies.
     /// </summary>
     public async Task<int> InstallDependenciesAsync(
@@ -38,21 +46,24 @@ public sealed class DependencyInstaller : IDisposable
         IProgress<InstallationProgress>? progress = null,
         CancellationToken cancellationToken = default)
     {
-        var dependencies = new List<(string Name, string Url, string Description)>
+        var dependencies = new List<(string Name, string Url, string LocalFile, string Description)>
         {
             (
                 "sp_WhoIsActive",
                 "https://raw.githubusercontent.com/amachanic/sp_whoisactive/refs/heads/master/sp_WhoIsActive.sql",
+                "sp_WhoIsActive.sql",
                 "Query activity monitoring by Adam Machanic (GPLv3)"
             ),
             (
                 "DarlingData",
                 "https://raw.githubusercontent.com/erikdarlingdata/DarlingData/main/Install-All/DarlingData.sql",
+                "DarlingData.sql",
                 "sp_HealthParser, sp_HumanEventsBlockViewer by Erik Darling (MIT)"
             ),
             (
                 "First Responder Kit",
                 "https://raw.githubusercontent.com/BrentOzarULTD/SQL-Server-First-Responder-Kit/refs/heads/main/Install-All-Scripts.sql",
+                "Install-All-Scripts.sql",
                 "sp_BlitzLock and diagnostic tools by Brent Ozar Unlimited (MIT)"
             )
         };
@@ -65,7 +76,7 @@ public sealed class DependencyInstaller : IDisposable
 
         int successCount = 0;
 
-        foreach (var (name, url, description) in dependencies)
+        foreach (var (name, url, localFile, description) in dependencies)
         {
             cancellationToken.ThrowIfCancellationRequested();
 
@@ -78,15 +89,40 @@ public sealed class DependencyInstaller : IDisposable
             try
             {
                 var depSw = Stopwatch.StartNew();
-                progress?.Report(new InstallationProgress { Message = $"[DEBUG] Downloading {name} from {url}", Status = "Debug" });
-                string sql = await DownloadWithRetryAsync(url, progress, cancellationToken: cancellationToken).ConfigureAwait(false);
-                progress?.Report(new InstallationProgress { Message = $"[DEBUG] {name}: downloaded {sql.Length} chars in {depSw.ElapsedMilliseconds}ms", Status = "Debug" });
+                string sql;
+
+                /* Check community/ directory first */
+                string? localPath = ResolveLocalFile(localFile);
+                if (localPath != null)
+                {
+                    progress?.Report(new InstallationProgress
+                    {
+                        Message = $"[DEBUG] {name}: loading from {localPath}",
+                        Status = "Debug"
+                    });
+                    sql = await File.ReadAllTextAsync(localPath, cancellationToken).ConfigureAwait(false);
+                }
+                else
+                {
+                    progress?.Report(new InstallationProgress
+                    {
+                        Message = $"[DEBUG] Downloading {name} from {url}",
+                        Status = "Debug"
+                    });
+                    sql = await DownloadWithRetryAsync(url, progress, cancellationToken: cancellationToken).ConfigureAwait(false);
+                }
+
+                progress?.Report(new InstallationProgress
+                {
+                    Message = $"[DEBUG] {name}: {(localPath != null ? "loaded" : "downloaded")} {sql.Length} chars in {depSw.ElapsedMilliseconds}ms",
+                    Status = "Debug"
+                });
 
                 if (string.IsNullOrWhiteSpace(sql))
                 {
                     progress?.Report(new InstallationProgress
                     {
-                        Message = $"{name} - FAILED (empty response)",
+                        Message = $"{name} - FAILED (empty {(localPath != null ? "file" : "response")})",
                         Status = "Error"
                     });
                     continue;
@@ -115,9 +151,10 @@ public sealed class DependencyInstaller : IDisposable
                     await command.ExecuteNonQueryAsync(cancellationToken).ConfigureAwait(false);
                 }
 
+                string source = localPath != null ? "local" : "GitHub";
                 progress?.Report(new InstallationProgress
                 {
-                    Message = $"{name} - Success ({description})",
+                    Message = $"{name} - Success ({description}) [{source}]",
                     Status = "Success"
                 });
 
@@ -156,6 +193,19 @@ public sealed class DependencyInstaller : IDisposable
         });
 
         return successCount;
+    }
+
+    /// <summary>
+    /// Checks the community directory for a local copy of the dependency file.
+    /// Returns the full path if found, null otherwise.
+    /// </summary>
+    private string? ResolveLocalFile(string fileName)
+    {
+        if (string.IsNullOrEmpty(_communityDirectory) || !Directory.Exists(_communityDirectory))
+            return null;
+
+        string path = Path.Combine(_communityDirectory, fileName);
+        return File.Exists(path) ? path : null;
     }
 
     private async Task<string> DownloadWithRetryAsync(

--- a/Installer/Program.cs
+++ b/Installer/Program.cs
@@ -636,7 +636,8 @@ namespace PerformanceMonitorInstaller
             Execute installation using Installer.Core
             Use DependencyInstaller for community dependencies before validation
             */
-            using var dependencyInstaller = new DependencyInstaller();
+            string communityDir = Path.Combine(monitorRootDirectory, "community");
+            using var dependencyInstaller = new DependencyInstaller(communityDir);
 
             var installResult = await InstallationService.ExecuteInstallationAsync(
                 connectionString,

--- a/InstallerGui/MainWindow.xaml.cs
+++ b/InstallerGui/MainWindow.xaml.cs
@@ -60,7 +60,8 @@ namespace PerformanceMonitorInstallerGui
             try
             {
                 InitializeComponent();
-                _dependencyInstaller = new DependencyInstaller();
+                string communityDir = System.IO.Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "community");
+                _dependencyInstaller = new DependencyInstaller(communityDir);
 
                 /*Set window title with version*/
                 Title = $"Performance Monitor Installer v{AppVersion}";

--- a/community/README.md
+++ b/community/README.md
@@ -1,0 +1,14 @@
+# Community Scripts (Offline Installation)
+
+Place pre-downloaded community SQL scripts in this directory for offline/air-gapped installations.
+When files are present here, the installer uses them instead of downloading from GitHub.
+
+## Expected files
+
+| File | Source | License |
+|------|--------|---------|
+| `sp_WhoIsActive.sql` | [amachanic/sp_whoisactive](https://github.com/amachanic/sp_whoisactive) | GPLv3 |
+| `DarlingData.sql` | [erikdarlingdata/DarlingData](https://github.com/erikdarlingdata/DarlingData/tree/main/Install-All) | MIT |
+| `Install-All-Scripts.sql` | [BrentOzarULTD/SQL-Server-First-Responder-Kit](https://github.com/BrentOzarULTD/SQL-Server-First-Responder-Kit) | MIT |
+
+Any file not found here will be downloaded from GitHub as usual.

